strings-csharp/ReadStringFromResourceFile/ReadStringFromResourceFile/Classes/ResourcesManager.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Globalization;
+using System.Reflection;
+using System.Resources;
+
+namespace ReadStringFromResourceFile;
+
+public class ResourcesManager
+{
+    private Assembly? assembly;
+    private ResourceManager resources;
+    private CultureInfo? culture;
+
+    public ResourcesManager(string resourceFile, Assembly? assembly = null, CultureInfo? cultureInfo = null)
+    {
+        assembly = assembly ?? Assembly.GetExecutingAssembly();
+
+        culture = cultureInfo ?? CultureInfo.CurrentCulture;
+        resources = new ResourceManager(resourceFile, assembly);
+
+        if (resources is null)
+            throw new Exception($"Resource file {resourceFile} not found.");
+    }
+
+    public string? GetString(string key)
+    {
+        try
+        {
+            var resource = GetResource<string>(key);
+            
+            if (!resource.IsValid)
+                return string.Empty;
```

Comment:
It seems inconsistent that we return empty string if the resource is not valid, but in an exception return null. I might be missing something here, so just want to mention it to think about it, whether this should be the case or not.